tests/queries/0_stateless/data_avro/union_in_complex_types.generate_avro.py

```diff
@@ -0,0 +1,59 @@
+import avro.schema
+import avro.datafile
+import avro.io
+import io
+
+# Define the schema
+schema = avro.schema.parse(
+    """
+    {
+      "type": "record",
+      "name": "TestRecord",
+      "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "union", "type": ["double", "string"]},
+        {"name": "union_with_null", "type": ["double", "string", "null"]},
+        {"name": "union_in_array", "type": {
+          "type": "array",
+          "items": ["double", "string"]
+        }},
+        {"name": "union_in_map", "type": {
+          "type": "map",
+          "values": ["double", "string"]
+        }}
```

Comment:
Could you, please, add unions with 3 or 4 not null fields? And try changing order of these fields in several ways to doublecheck that there are no any problems with that global-and-local-discriminator stuff, please